src/engine/sim_engine/pool.rs

```diff
@@ -81,6 +81,12 @@ impl Pool for SimPool {
         destroy_filesystems!{self; fs_names}
     }
 
+    fn destroy(mut self) -> EngineResult<()> {
+        self.block_devs.clear();
```

Comment:
Why is clearing these data structures necessary for SimPool and not for StratPool?